### ISSUE TITLE
Players Moving With Vehicles

### ISSUE THREAD

**Issue by Nianor** [DonBruce64/MinecraftTransportSimulator#9]
Could we get players and entities on vehicles to move with them? It would be nice. And fun. And I wanna try putting a car onto a plane.

**Comment by DonBruce64** [comment]
Uuuh, players *already* do this as of V11.2.0.  There's even a JSON parameter for making it so players don't get blown off by wind shear which is specially designed with zepplins and fun car action scenes in mind.

**Comment by Nianor** [comment]
...I tested this before posting. It wasn't working for me, unless I wasn't understanding something about how the system works. Or unless you fall off external surfaces at *really* low speeds.

**Comment by DonBruce64** [comment]
You shouldn't.  What MTS version was this in?

**Comment by Nianor** [comment]
V11.2.0, as you said.

**Comment by DonBruce64** [comment]
Odd.  You wouldn't happen to have a video, would you?

**Comment by Nianor** [comment]
Only unusual aspect about it was that I _was_ running it in my dev environment. I'm not doing anything that should screw with MTS physics or hit detection, but I guess it's possible.

**Comment by Nianor** [comment]
I can try to recreate and get video, sure.

**Comment by DonBruce64** [comment]
Can you try in regular MC?  You never know with dev what gets fouled up.

**Comment by Nianor** [comment]
Let me check in my dev environment first. It's possible something was just wonky, or I messed up testing. If it's still an issue, I'll confirm in my normal directory.

**Comment by Nianor** [comment]
Yeah, it looks like the issue isn't with the player not being able to stand on the hitboxes, it's with them not keeping up with the vehicle. They start falling off backwards at really low speeds. Like, "slower than walking" speeds.

**Comment by DonBruce64** [comment]
What's your speed factor, and can you put an BLK/S gauge in the vehicle or give me a rough calculation of the exact speed?

**Comment by Nianor** [comment]
Speed factor is untouched. As far as speed, it looks to be about 3.5 BLK/S.

**Comment by DonBruce64** [comment]
That doesn't make sense... in the code you don't start slipping until you hit at least 5 BLK/S.  Is this on a large or small hitbox?

**Comment by Nianor** [comment]
...I have a video, don. XD

**Comment by Nianor** [comment]
It takes a while before the falling off bit starts, but it also seems to start kinda gradually. It started way before 3.5 BLK/S, that was just the point where I couldn't easily stay standing on the wing any more.

**Comment by DonBruce64** [comment]
Will investigate the slippage.  That does indeed look like too much for that speed.  I may need to tweak something there...

**Comment by DonBruce64** [comment]
I know what this is.  I'm calculating the 5 BLK/S using the un-factored speed that's internal to the vehicle, not the factored speed that the actual vehicle is moving at.  This means that when I check for 5 BLK/S at 0.35 speed factor it's actually at 0.75 BLK/S.  Will fix.

**Comment by Nianor** [comment]
Awesome.

**Comment by Nianor** [comment]
Can it be configurable? :3

**Comment by DonBruce64** [comment]
Yes.  But that's a separate ticket. :)

**Comment by DonBruce64** [comment]
Fixed in Commit 78181d86a6dd4cbf58a981e99a275453b37d12be.
